src/slurm_plugin/slurm_resources.py

```diff
@@ -168,9 +167,7 @@ def is_drained(self):
 
     def is_powering_down(self):
         """Check if slurm node is in powering down state."""
-        return self.SLURM_SCONTROL_POWERING_DOWN_STATES in self.states or (
-            self.SLURM_SCONTROL_POWERED_STATE not in self.states and self.SLURM_SCONTROL_POWER_DOWN_STATE in self.states
-        )
+        return self.SLURM_SCONTROL_POWERING_DOWN_STATE in self.states
```

Comment:
is_powering_down() is also used in checking unhealthy nodes:
```
    def is_down(self):
        """Check if slurm node is in a down state."""
        return self.SLURM_SCONTROL_DOWN_STATE in self.states and not self.is_powering_down()
```
With this change, dynamic nodes state with (down+power_down) and (drain+power_down) will be treated as unhealthy dynamic node,  instances will be terminated and nodes wil be power_down_force by `_handle_unhealthy_dynamic_nodes`

Same for static nodes not in replacement
Should we also modify the is_state_healthy() function?

Reply:
I'm going to add a commit to not consider the node in DRAIN+POWER_DOWN as drained, so to give Slurm time to transition the node from POWER_DOWN to POWERING_DOWN by itself.
Instead for the DOWN+POWER_DOWN, I think it's OK to consider it as unhealthy.